webrtc/src/peer_connection/sdp/mod.rs

```diff
@@ -505,16 +505,8 @@ pub(crate) async fn add_transceiver_sdp(
         return Ok((d, false));
     }
 
-    let mut directions = vec![];
-    if t.sender().await.is_some() {
-        directions.push(RTCRtpTransceiverDirection::Sendonly);
-    }
-    if t.receiver().await.is_some() {
-        directions.push(RTCRtpTransceiverDirection::Recvonly);
-    }
-
     let parameters = media_engine
-        .get_rtp_parameters_by_kind(t.kind, &directions)
+        .get_rtp_parameters_by_kind(t.kind, t.direction())
```

Comment:
This is subtly wrong and should be different when we are generating an answer I think, see the code at line 594

Reply:
What is the RTCRtpTransceiverDirection on transceiver then?

Is it storing the mirrored or straight dir?

Reply:
I'm thinking like this:

If we've added an extension to MediaEngine as `SendOnly`
We are figuring out which extensions to use, and the transceiver is also `SendOnly` or `Sendrecv`.

This should match. I.e. unless transceiver direction is stored mirrored, none of these should be inverted.

Reply:
Notice that we never communicate the direction. Even though the SDP could we always just create a directionless entry in the SDP.

```
        media = media.with_extmap(sdp::extmap::ExtMap {
            value: rtp_extension.id,
            uri: Some(ext_url),
            ..Default::default()
        });
```